File: aula3/cpp_logic/calculadora.cpp
#include <iostream>
using namespace std; // Evita ter que escrever std:: toda hora

int main() {
    // Alocando variáveis na memória antes de usar (característica de linguagens robustas)
    double num1, num2;
    
    cout << "--- Calculadora Basica ---" << endl;
    cout << "Digite o primeiro numero: ";
    cin >> num1; // Extrai o valor digitado para a variável
    
    cout << "Digite o segundo numero: ";
    cin >> num2;
    
    cout << "Soma: " << (num1 + num2) << endl;
    cout << "Subtracao: " << (num1 - num2) << endl;
    cout << "Multiplicacao: " << (num1 * num2) << endl;
    
    // Tratamento sequencial contra quebra por divisão por zero
    if (num2 == 0) {
        cout << "Erro: Nao e possivel realizar divisao por zero!" << endl;
    } else {
        cout << "Divisao: " << (num1 / num2) << endl;
    }
    
    return 0; // Indica pro SO que a execução ocorreu sem erros
}

File: aula3/java_logic/Calculadora.java
import java.util.Scanner;

public class Calculadora {
    public static void main(String[] args) {
        // Instanciando o Scanner para ler as entradas do usuário
        Scanner entrada = new Scanner(System.in);
        
        System.out.println("--- Calculadora Basica ---");
        System.out.print("Digite o primeiro numero: ");
        double num1 = entrada.nextDouble(); // Tipagem estática exigida pelo Java
        
        System.out.print("Digite o segundo numero: ");
        double num2 = entrada.nextDouble();
        
        System.out.println("Soma: " + (num1 + num2));
        System.out.println("Subtracao: " + (num1 - num2));
        System.out.println("Multiplicacao: " + (num1 * num2));
        
        // Validação com if/else para proteger a memória de erros matemáticos
        if (num2 == 0) {
            System.out.println("Erro: Nao e possivel realizar divisao por zero!");
        } else {
            System.out.println("Divisao: " + (num1 / num2));
        }
        
        entrada.close(); // Liberando recursos da memória
    }
}